### PR TITLE
Device event data api

### PR DESCRIPTION
**Pull request by rikkegammelmark**

### CODE REVIEW

File: proto/toit/model/device.proto [hunk 1]
```diff
@@ -121,3 +121,72 @@ enum DeviceSessionChangeType {
...
+  google.protobuf.Timestamp created = 3;
+  Type type = 4;
+  string msg = 5;
+  bytes data = 6;
+  Initiater initiater = 7;
+
+  repeated Info information = 8;
+}
+
+message Initiater {
```
Comment (jstroem) [inline proto/toit/model/device.proto:147]: consider moving Info and Intiater into DeviceEvents. They are only revelent within that message.

File: proto/toit/model/device.proto [hunk 2]
```diff
@@ -121,3 +121,72 @@ enum DeviceSessionChangeType {
...
+  }
+
+  oneof initiater {
+    Device device = 1;
+    Console console = 2;
+    User user = 3;
+  };
+}
+
+message Info {
```
Comment (jstroem) [inline proto/toit/model/device.proto:166]: nit: rename to KV?

File: proto/toit/model/device.proto [hunk 3]
```diff
@@ -121,3 +121,72 @@ enum DeviceSessionChangeType {
...
+  string key = 1;
+  Value value = 2;
+
+  message Value {
+
+    message Simple {
```
Comment (jstroem) [inline proto/toit/model/device.proto:172]: Move out of value and rename to PrimitiveValue

File: proto/toit/model/device.proto [hunk 4]
```diff
@@ -121,3 +121,72 @@ enum DeviceSessionChangeType {
...
+       enum Type {
+        JSON = 0;
+        STRING = 1;
+      }
+
+      Type type = 1;
+      string value = 2;
+    }
+
+    message Diff {
```
Comment (jstroem) [inline proto/toit/model/device.proto:182]: rename to DiffValue

File: proto/toit/model/device.proto (hunk 2 above)
Comment (rikkegammelmark) [reply proto/toit/model/device.proto:166]: I kind of like Info, I will leave it for now.

**Comment by rikkegammelmark** [comment]
I agree that nesting does not look too good in Go, it works pretty well in Typescript though, and I think it makes the api more readable. I will keep the nesting for now.

File: proto/toit/model/device.proto (hunk 3 above)
Comment (rikkegammelmark) [reply proto/toit/model/device.proto:172]: I think the nesting reads better, so I will leave it as it is.

File: proto/toit/model/device.proto (hunk 4 above)
Comment (rikkegammelmark) [reply proto/toit/model/device.proto:182]: Same as above.

**Pull request merged**